src/fixate/drivers/dmm/keithley_6500.py

```diff
@@ -0,0 +1,299 @@
+        if errors:
+            if silent:
+                return errors
+            else:
+                raise InstrumentError(
+                    "Error(s) Returned from DMM\n"
+                    + "\n".join(
+                        [
+                            "Code: {}\nMessage:{}".format(code, msg)
+                            for code, msg in errors
+                        ]
+                    )
+                )
+
+    def error_cleanup(self):
+        """
+        When VisaIOError exception caught, DMM interrupt is sent, read buffer is cleared and DMM returned to power up
+        state.  VI read buffer is then flushed.
+        DMM is then returned to previous configuration
+        """
+        self._CLEAN_UP_FLAG = True
+        # Disaster Recovery
+        self.instrument.write("\x03;*RST;*CLS")  # CTRL-C
+        time.sleep(1.1)  # time needed to clear the dmm read buffer
+        self.instrument.flush(constants.VI_READ_BUF_DISCARD)
+        self.instrument.close()
+        self.instrument.open()
+
+    def _set_measurement_mode(self, mode, _range=None, suffix=None):
+        """
+        Helper function used to set the measurement mode for voltage_ac, voltage_dc, current_ac, current_dc,
+        resistance, fresistance. Reduces previous duplicate code.
+        :param mode:
+        :param _range:
+        :return:
+        """
+        self.mode = mode
+        mode_str = "SENS:FUNC '{}'".format(self._modes[self._mode])
+        if _range is not None:
+            mode_str += "; :SENS:{}:RANGE {}".format(self._modes[self._mode], _range)
+        if suffix is not None:
+            mode_str += " {}".format(suffix)
+        self._write(mode_str)
+        # Make sure sample count is set for the mode:
+        self._write(["SENS:COUN {}".format(self.samples)])
+
+        self._is_error()
+
+    def voltage_ac(self, _range=None):
+        self._set_measurement_mode("voltage_ac", _range)
+
+    def voltage_dc(self, _range=None):
+        self._set_measurement_mode("voltage_dc", _range)
+
+    def current_ac(self, _range=None):
+        self._set_measurement_mode("current_ac", _range)
+
+    def current_dc(self, _range=None):
+        self._set_measurement_mode("current_dc", _range)
+
+    def resistance(self, _range=None):
+        self._set_measurement_mode("resistance", _range)
+
+    def fresistance(self, _range=None):
+        self._set_measurement_mode("fresistance", _range)
+
+    def frequency(self, _range=None):
+        # Cannot set range for frequency measurement
+        self._set_measurement_mode("frequency")
+
+    def period(self, _range=None):
+        # Cannot set range for period measurement
+        self._set_measurement_mode("period")
+
+    def capacitance(self, _range=None):
+        self._set_measurement_mode("capacitance", _range)
+
+    def diode(self, low_current=True, high_voltage=False):
+        # Cannot set range. 10V fixed range
+        self._set_measurement_mode("diode")
```

Comment:
Is the single keithley diode test equivalent to the flukes high_current + high_voltage? There's a few zener tests in test scripts that rely on that.
Different current modes may also affect diode Vf test values?

Reply:
I have fixed this up. Found a command to set the bias current, but I think the range (high/low voltage) setting has to remain fixed. 
I assume its primarily the bias current that matters the most? Testing on the jig I have gives better measurements now.

Reply:
If each modes bias currents are the same and the keithley's default voltage is ≥ the fluke's when high_voltage=True then it should be fine

Reply:
Diode bias current behavior should be normalized between the two drivers.